### PR TITLE
ZCS-1033 external account shows 500 error when password restrictions not met

### PR DESCRIPTION
**Pull request by rohan-ambasta**
[Issue]:
This bug occurred because ServletException was thrown in case of any exception while creating the virtual  account. 
[Fix]:
Handle the exception and send the error code and error message to extuserprov.jsp. I will create a separate JIRA ticket to handle the UI change
[Testing]:
Manual testing

### CODE REVIEW

File: store/src/java/com/zimbra/cs/service/ExternalUserProvServlet.java [hunk 1]
```diff
@@ -362,8 +416,10 @@ private static void provisionVirtualAccountAndRedirect(HttpServletRequest req, H
             enableAppFeatures(grantee, viewTypes);
 
             setCookieAndRedirect(req, resp, grantee);
-        } catch (Exception e) {
-            throw new ServletException(e);
+        } catch (ServiceException e) {
+            throw e;
+        }  catch (Exception e) {
+            throw ServiceException.TEMPORARILY_UNAVAILABLE();
```
Comment (rupalid) [inline store/src/java/com/zimbra/cs/service/ExternalUserProvServlet.java]: Log the exception at debug level

Comment (rohan-ambasta) [inline store/src/java/com/zimbra/cs/service/ExternalUserProvServlet.java]: done

File: store/src/java/com/zimbra/cs/service/ExternalUserProvServlet.java [hunk 2]
```diff
@@ -277,19 +278,70 @@ protected void doPost(HttpServletRequest req, HttpServletResponse resp) throws S
...
+            sendHtmlResponse(resp, reqHeaders);
+        } else {
+            for (Map.Entry<String, String> entry : errorAttrs.entrySet()) {
+                req.setAttribute(entry.getKey(), entry.getValue());
+            }
+            forward(req, resp);
+        }
+    }
+
+    private void forward(HttpServletRequest req, HttpServletResponse resp) throws ServletException, IOException {
```
Comment (shrikant-prasad) [inline store/src/java/com/zimbra/cs/service/ExternalUserProvServlet.java]: similar code is there at several places in this class, better to make the jsp page as argument of the method and use this method at all these places to remove duplicate code. same for sendHtmlResposne method.

Comment (rohan-ambasta) [inline store/src/java/com/zimbra/cs/service/ExternalUserProvServlet.java]: done

**Pull request closed without merging**